apps/jobs/views.py

```diff
@@ -382,12 +381,12 @@ def change_submission_data_and_visibility(
             if (
                 challenge_phase.is_restricted_to_select_one_submission
                 and is_public
-                and submissions_already_public
+                and submissions_already_public.count() == 1
             ):
-                submissions_already_public.is_public = False
-                submissions_already_public.save()
-        except Submission.DoesNotExist:
-            pass
+                # Assuming only 1 submission would be public
+                for submission_already_public in submissions_already_public:
```

Comment:
We don't need a loop here, right? Since we know this list will only have 1 entry we can simply use the array indexing to parse it through the serializer.

Reply:
i tried using index but it wasn't reflecting in db seems like a access by reference issue that's why used for loop here